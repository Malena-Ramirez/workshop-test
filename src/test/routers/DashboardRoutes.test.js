import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { mount, configure } from 'enzyme';
import { AuthContext } from '../../auth/AuthContext';
import { MemoryRouter } from 'react-router-dom';
import { DashboardRoutes } from '../../routers/DashboardRoutes';

configure({ adapter: new Adapter() });

describe('Prueba con componente <DashboardRoutes/>', () => {
  const contextValue = {
    dispatch: jest.fn(),
    user: {
      logged: true,
      name: 'Isaias'
    }
  }

  test('Verificar el nombre del usuario registrado', () => {
    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        <MemoryRouter>
          <DashboardRoutes />
        </MemoryRouter>
      </AuthContext.Provider>
    )
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.text-info').text().trim()).toBe('Isaias');
  })

})

