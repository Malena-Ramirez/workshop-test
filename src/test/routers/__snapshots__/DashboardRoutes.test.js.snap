// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba con componente <DashboardRoutes/> Verificar el nombre del usuario registrado 1`] = `ReactWrapper {}`;
